### PR TITLE
Fix fs bad encryption revision in sync monitor

### LINKED ISSUE #1725: FSBadEncryptionRevision: Cannot fetch vlob e2b92057-3254-4a95-8479-3b8807f3c98b: Bad encryption revision provided
Sentry Issue: [PARSEC-QWB](https://sentry.io/organizations/scille/issues/2391494688/?referrer=github_integration)

```
FSBadEncryptionRevision: Cannot fetch vlob e2b92057-3254-4a95-8479-3b8807f3c98b: Bad encryption revision provided
(1 additional frame(s) were not displayed)
...
  File "parsec\core\sync_monitor.py", line 217, in tick
    await self._sync(entry_id)
  File "parsec\core\sync_monitor.py", line 283, in _sync
    await self.workspace.sync_by_id(entry_id, recursive=False)
  File "parsec\core\fs\workspacefs\workspacefs.py", line 644, in sync_by_id
    manifest = await self._sync_by_id(entry_id, remote_changed=remote_changed)
  File "parsec\core\fs\workspacefs\workspacefs.py", line 565, in _sync_by_id
    remote_manifest = await self.remote_loader.load_manifest(entry_id)
  File "parsec\core\fs\remote_loader.py", line 445, in load_manifest
    f"Cannot fetch vlob {entry_id}: Bad encryption revision provided"

{'workspace_id': <EntryID 27180f0b048b41598bebb68eb23d5385>, 'event': 'Sync monitor has crashed', 'logger': 'parsec.core.sync_monitor', 'level': 'exception', 'timestamp': '2021-05-10 19:07:22'}
```

### PR DESCRIPTION
**Pull request by touilleMan**
fix #1725

### CODE REVIEW

File: parsec/core/sync_monitor.py [hunk 1]
```diff
@@ -257,8 +261,11 @@ async def tick(self) -> float:
                     # the write access in the future) but pretent it just accured
                     # to avoid a busy sync loop until `read_only` flag is updated.
                     self._local_changes[entry_id] = LocalChange(now)
-                except FSWorkspaceInMaintenance:
-                    # Not the right time for the sync, retry later
+                except (FSWorkspaceInMaintenance, FSBadEncryptionRevision):
+                    # Not the right time for the sync, retry later.
+                    # `FSBadEncryptionRevision` occurs if the reencryption is quick
+                    # enough to start and finish before we process the sharing.reencrypted
+                    # message so we try a sync with the old encryption revision.
```
Comment (vxgmichel) [inline parsec/core/sync_monitor.py:268]: This fix seems OK, but I'm a bit worried that we can no longer tell the difference between an acceptable race condition (accessing the realm while the new encryption key hasn't been received) and bug in the reencryption logic. Is there a way to tell those appart and maybe log the latter?

**Review by vxgmichel — APPROVED** [review]
LGTM :+1:

Just a few comments.

File: tests/core/backend_connection/test_authenticated_conn.py [hunk 2]
```diff
@@ -123,6 +123,9 @@ async def _bad_monitor(*, task_status=trio.TASK_STATUS_IGNORED):
                 {"status": BackendConnStatus.CRASHED, "status_exc": spy.ANY},
             )
             assert conn.status == BackendConnStatus.CRASHED
+            caplog.assert_occured(
+                "[exception] Unhandled exception            [parsec.core.backend_connection.authenticated]"
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply tests/core/backend_connection/test_authenticated_conn.py:127]: yes totally agree, I though about improving `_remove_colors` to also trip the spaces, but it seems a bit complicated to do so I consider we will handle this in another PR when we will hit the issue (typically if updating structlog version breaks the tests)

File: tests/core/test_sync_monitor.py [hunk 3]
```diff
@@ -386,3 +386,53 @@ async def test_sync_monitor_while_changing_roles(
...
+    assert not alice_core.are_monitors_idle()
+
+    # Now let Alice process the `sharing.reencrypted` messages, this should
+    # allow to do the sync
+    allow_message_processing.set()
+    with trio.fail_after(60):  # autojump, so not *really* 60s
+        await alice_core.wait_idle_monitors()
+
+    # Just make sure the sync is done
+    await bob_workspace.sync()
```
[the remark this answers is not in the document]
Comment (touilleMan) [reply tests/core/test_sync_monitor.py:434]: Alice should have modified the workspace as soon as `allow_message_processing.set()` is called, so bob must sync to see Alice's changes

**Pull request merged**